### PR TITLE
Internalise connection context

### PR DESCRIPTION
**Pull request by jbowkett**
All green on CI

### CODE REVIEW

File: mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseConnectionContext.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.env.CouchbaseEnvironment;
+import com.excelian.mache.builder.storage.AbstractConnectionContext;
+import com.excelian.mache.core.AbstractCacheLoader;
+
+import java.util.List;
+
+/**
+ * Created by jbowkett on 21/01/2016.
```
Comment (tellisnz) [inline mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseConnectionContext.java]: Uncle bob is crying.

File: mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseProvisioner.java [hunk 2]
```diff
@@ -97,17 +67,27 @@ public static ClusterBuilder couchbase() {
      * A builder with defaults for a Couchbase cluster.
      */
     public static class CouchbaseProvisionerBuilder {
-        private final ConnectionContext<Cluster> connectionContext;
         private final BucketSettings bucketSettings;
+        private CouchbaseEnvironment couchbaseEnvironment = DefaultCouchbaseEnvironment.create();
```
Comment (tellisnz) [inline mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseProvisioner.java]: Actually this is bad - we may need to force users to provide a couchbase env - if we always create it here and then the user creates their own, they're always going to get the warning in the logs. maybe add this one to the mandated props, or in the create below, have if null then env.create?

**Comment by tellisnz** [comment]
Apart from above :+1:  - shotgun my PR going first though! :D

File: mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseConnectionContext.java (hunk 1 above)
Comment (jbowkett) [inline mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseConnectionContext.java]: I know.  Bad times.

File: mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseProvisioner.java (hunk 2 above)
Comment (jbowkett) [inline mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseProvisioner.java]: Let's only initialise if it to default if it is null when we need it.

Ask don't tell.

**Pull request merged**